src/OrchardCore.Modules/OrchardCore.Lists/Views/ListPart.DetailAdmin.cshtml

```diff
@@ -42,7 +42,7 @@ else
                     <div class="dropdown-menu" aria-labelledby="dropdownMenuLink">
                         @foreach (var containedContentTypeDefinition in Model.ContainedContentTypeDefinitions)
                         {
-                            <a class="dropdown-item" asp-action="Create" asp-controller="Admin" asp-route-id="@containedContentTypeDefinition.Name" asp-route-area="OrchardCore.Contents" asp-route-ListPart.ContainerId="@Model.ListPart.ContentItem.ContentItemId" asp-route-ListPart.EnableOrdering="@Model.EnableOrdering">@containedContentTypeDefinition.DisplayName</a>
+                            <a class="dropdown-item" asp-action="Create" asp-controller="Admin" asp-route-id="@containedContentTypeDefinition.Name" asp-route-area="OrchardCore.Contents" asp-route-ListPart.ContainerId="@Model.ListPart.ContentItem.ContentItemId" asp-route-ListPart.EnableOrdering="@Model.EnableOrdering" asp-route-returnUrl="@FullRequestPath">@containedContentTypeDefinition.DisplayName</a>
```

Comment:
Cool, this looks good. 

It also needs to be added to the Single Create button 20 lines above (as this is only for if multiple content types are creatable.

Reply:
Hey Dean,

Thanks for taking a look. In my case I had multiple contained items. I have added the return url to the other statement for single contained items and ran some quick validation tests to ensure everything is still good